File: client/src/app/modules/miscellaneous/services/guichet-unique/type.voie.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { TypeVoie } from 'src/app/modules/quotation/model/guichet-unique/referentials/TypeVoie';
import { AppRestService } from 'src/app/services/appRest.service';

@Injectable({
  providedIn: 'root'
})
export class TypeVoieService extends AppRestService<TypeVoie>{

  constructor(http: HttpClient) {
    super(http, 'quotation/guichet-unique');
  }

  getTypeVoie() {
    return this.getListCached(new HttpParams(), 'type-voie');
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/guichetUnique/referentials/TypeVoieService.java
package com.jss.osiris.modules.osiris.quotation.service.guichetUnique.referentials;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.guichetUnique.referentials.TypeVoie;

public interface TypeVoieService {
    public List<TypeVoie> getTypeVoie();

    public TypeVoie getTypeVoie(String code);
}
